storage_service/locations/models/package.py

```diff
@@ -1237,15 +1237,26 @@ def create_pointer_file(self,
         for agent in premis_agents:
             mets_fs_entry.add_premis_agent(agent.serialize())
         pointer_file.append_file(mets_fs_entry)
-        # 3. Validate the pointer file
+        # Validate the pointer file
         if validate:
-            is_valid, report = metsrw.validate(
-                pointer_file.serialize(), schematron=metsrw.AM_PNTR_SCT_PATH)
-            if is_valid:
-                LOGGER.info('Pointer file constructed for %s is valid.', self.uuid)
-            else:
-                LOGGER.warning('Pointer file constructed for %s is not valid.\n%s',
-                            self.uuid, metsrw.report_string(report))
+            try:
+                is_valid, report = metsrw.validate(
+                    pointer_file.serialize(),
+                    schematron=metsrw.AM_PNTR_SCT_PATH)
+            except etree.XMLSchemaParseError as err:
+                # It has been observed that this function can fail validating,
+                # for example against the Library of Congress' (LoC) XSD for
+                # METS if the LoC server is down it will fail. We don't want
+                # this to happen so if the pointer file xml is not valid at
+                # this point in time we need to rely on a different mechanism
+                # to manage that.
+                LOGGER.warning("Failed to parse validation schema: %s", err)
+                return pointer_file
+            if not is_valid:
```

Comment:
Not sure if I'm missing something here but isn't this going to raise a `NameError` since `is_valid` is undefined when   `etree.XMLSchemaParseError` is raised?

Reply:
I'm expecting `is_valid` to either be declared and then an error to be raised and then we return. `is_valid` remains in scope when we test `if not is_valid`. 

That's not to say it is correct @sevein but if there is potential for something to go wrong here then perhaps a better way to approach this is in the https://github.com/archivematica/issues log with with a complete bug report? This PR was merged six months back.

Reply:
I wasn't sure if it was a bug!
I think it was an oversight on my part, you're returning inside the except.
I've no idea why I landed into this PR, I was doing some tidying yesterday and realized I missed this change.
Thanks! :heart_eyes:

Reply:
Fair dos @sevein!! You know previously this snippet from me would have been far more verbose somehow - but someone taught me about the happy path :wink: (which I think (i hope!) this might be a small demonstration of :smile:)